tiledb/sm/subarray/subarray.cc

```diff
@@ -1313,13 +1455,141 @@ TileOverlap Subarray::get_tile_overlap(uint64_t range_idx, unsigned fid) const {
 void Subarray::swap(Subarray& subarray) {
   std::swap(array_, subarray.array_);
   std::swap(layout_, subarray.layout_);
+  std::swap(cell_order_, subarray.cell_order_);
   std::swap(ranges_, subarray.ranges_);
   std::swap(is_default_, subarray.is_default_);
   std::swap(range_offsets_, subarray.range_offsets_);
   std::swap(tile_overlap_, subarray.tile_overlap_);
   std::swap(est_result_size_computed_, subarray.est_result_size_computed_);
   std::swap(tile_overlap_computed_, subarray.tile_overlap_computed_);
   std::swap(est_result_size_, subarray.est_result_size_);
+  std::swap(max_mem_size_, subarray.max_mem_size_);
+  std::swap(relevant_fragments_, subarray.relevant_fragments_);
+}
+
+Status Subarray::compute_relevant_fragments() {
+  STATS_START_TIMER(stats::Stats::TimerType::READ_COMPUTE_RELEVANT_FRAGS)
+
+  auto meta = array_->fragment_metadata();
+  auto fragment_num = meta.size();
+  auto range_num = this->range_num();
+  std::vector<uint8_t> frag_bitmap(fragment_num, 0);
+
+  // Compute the relevant fragments
+  auto statuses = parallel_for_2d(
```

Comment:
`statuses` appears to be unchecked

Reply:
That's because each loop iteration always returns `Status::Ok()`.